clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -128,13 +128,27 @@ public UUIDSerde() {
 
     static public final class ListSerde<Inner> extends WrapperSerde<List<Inner>> {
 
+        final static int NEGATIVE_SIZE_VALUE = -1;
```

Comment:
Should it be parametrized?

Reply:
What is this? Can you give it a name that describes what it means a little more -- IIUC this is a sentinel that indicates "this list has variable-sized elements so we encode each element's size".

That said, coming up with names is hard -- you can probably do a better job than me but just to throw out a suggestion, what about something like `VARIABLE_SIZE_SENTINEL`?

Reply:
Basically, if we are following `VARIABLE_SIZE` serialization strategy **and** we have a `null` entry in our list, we encode this null entry as `-1`, so that during deserialization when we encounter `-1`, we append `null` entry to our list.

Example, to serialize a list like `{"A", "B", null, "C"}` of strings, the payload would look smth like:
```
serialization_flag | size_of_list_4 | A_size | A_encoded | B_size | B_encoded | -1 | C_size | C_encoded
```

Reply:
Could be called something like `NULL_ENTRY_VALUE` instead maybe?

Reply:
Ooooh ok, that makes a lot more sense now. I think your suggestion for the name sounds good